### DISCUSSION TITLE
Failed to load resource: the server responded with a status of 404 ()

### DISCUSSION THREAD

**Discussion by python2020-maker** [jspsych/jsPsych#1769] [Q&A]
Hi, 

I can run the experiment but the data is not saved and this error in console: Failed to load resource: the server responded with a status of 404 ()

How can this issue be solved? Is there anybody to help?

**Comment by jodeleeuw** [comment]
Hi @python2020-maker,

We'll need more information to be able to help. Where are you running your experiment? How is the data being saved? What file is the 404 error referencing?

**Comment by becky-gilbert** [comment]
Hi @python2020-maker, thanks for sharing the link to your task. It looks like you're using an old version of jsPsych, prior to v6.0.5. Versions of jsPsych <6.0.5 aren't supported on cognition.run, even when you upload all of the old files - see [this comment](https://github.com/jspsych/jsPsych/discussions/1709#discussioncomment-573468) from the cognition.run creator on a different discussion thread.

As far as the 404 errors - it looks like you're using the single-stim plugin with an HTML stimulus, and jsPsych is trying to preload the HTML. Obviously there are no files named with your HTML strings, so that's why it's failing. I'm not sure why that's happening, but if you're not using any images, you could comment out the `registerPreload` line in the plugin file and see if that helps.

![single_stim_preload](https://user-images.githubusercontent.com/9041788/116325865-0b27bd80-a778-11eb-9b20-ee7640459ed2.PNG)

Another thing is that you're putting the stimulus into an array, when I think it should just be a string. like this:
```js
var practice_block14 = {
  type: "single-stim",
  stimulus: '<p>music</p>',
  choices: [100], 
  timing_response: 1200, 
  data: {phase: 'mem'},
  is_html: true,
};
```
I don't think it would be too difficult to make your expeirment compatible with jsPsych v6.0.5+ so that you can run it on cognition.run. For all of your single stim trials that use an HTML stimulus, you can switch to the html-keyboard-response plugin (see [plugin docs](https://www.jspsych.org/plugins/jspsych-html-keyboard-response/) for changes to parameter names, etc.). Another big change in v6.3.0+ is the switch from key codes to key characters, so instead of `key_forward: 13` or `choices: [32]`, you would use `key_forward: 'Enter'` or `choices: [' ']`. There are lots of other changes, which you can see in the [release notes for each release](https://github.com/jspsych/jsPsych/releases) - just start with the version you're using now and look at all of the changes between then and the version you're switching to.

**Reply by python2020-maker** [reply]
Thank you so much.
